src/LibraryUtilities.ts

```diff
@@ -566,48 +285,244 @@ export function updateSections(oldLayoutSpecs: any, newLayoutSpecs: any): void {
- * by splitting the name with '.', and put them under a specific section.
+ * This function will convert layoutElements to ItemData, and construct an array of IncludeItemPair,
+ * which stores an includeInfo and its parent item.
  * 
- * As an example, for two typeListNodes 'A.B.C' and 'A.B.D', the resulting nodes are:
- * - A
- *   |- B
+ * @param {LayoutElement[]} layoutElements 
+ * The layoutElements to be converted to ItemData
+ * 
+ * @param {IncludeItemPair[]} pairs
+ * The IncludeItemPairs to be contructed from
+ * 
+ * @param {ItemData} parentItem 
+ * The parentItem of layoutElements (if exist)
+ */
+function convertLayoutElementToItemData(
+    layoutElements: LayoutElement[],
+    pairs: IncludeItemPair[],
+    parentItem?: ItemData):
+    ItemData[] {
+
+    let results: ItemData[] = [];
+
+    for (let layoutElement of layoutElements) {
+        let layoutData = new ItemData(layoutElement.text);
+        layoutData.constructFromLayoutElement(layoutElement);
+
+        // Sections that don't show its header will be expanded by default.
+        if (layoutData.itemType === "section" && !layoutData.showHeader) {
+            layoutData.expanded = true;
+        }
+
+        if (parentItem) {
+            parentItem.appendChild(layoutData);
+        }
+
+        if (layoutElement.include.length > 0) {
+            for (let include of layoutElement.include) {
+                let pair = new IncludeItemPair();
+                pair.include = include;
+                pair.parentItem = layoutData;
+                pairs.push(pair);
+            }
+        }
+
+        if (layoutElement.childElements.length > 0) {
+            convertLayoutElementToItemData(layoutElement.childElements, pairs, layoutData);
+        }
+
+        results.push(layoutData);
+    }
+
+    return results;
+}
+
+/**
+ * This method will merge all typeListNodes under corresponding library items based on includeItemPairs.
+ * Each includeItemPair stores includeInfo and the parent item of this includeInfo.
+ * 
+ * When iterating through includeItemPairs, comparison will happen between fullyQualifiedName of typeListNode 
+ * and path of the include info of each pair.. 
+ * 
+ * If there is a match of them, which means they are exactly the same, this type node will be merged to 
+ * the parent item of the includeInfo. 
+ * 
+ * If there is not a match, but fullyQualifiedName includes path, this type node will
+ * first be built from name, and then merged to the parent item.
+ * For example, if fullayQualifiedName is A.B.C.D, path is A.B, the item to be merged would have the 
+ * following structure.
+ * 
+ * - parentItem
+ *  |- B
  *      |- C
- *      |- D
+ *          |- D
  * 
- * @param {TypeListNode[]} allNodes 
- * All the typeListNodes
+ * If there is not a match or include, skip to next includeInfo/typeListNode, depending on their 
+ * lexicographical order.
```

Comment:
Please don't forget to update this comment after you've renamed `IncludeItemPair` later.